astroquery/desi/core.py

```diff
@@ -0,0 +1,111 @@
+        """
+
+        tap_service = vo.dal.TAPService(conf.tap_service_url)
+        coordinates_transformed = coordinates.transform_to(coord.ICRS)
+
+        qstr = (f"SELECT all * FROM ls_dr{data_release}.tractor WHERE "
+               f"dec>{(coordinates_transformed.dec - radius).to_value(u.deg)} and "
+               f"dec<{(coordinates_transformed.dec + radius).to_value(u.deg)} and "
+               f"ra>{coordinates_transformed.ra.to_value(u.deg) - radius.to_value(u.deg) / np.cos(coordinates_transformed.dec)} and "
+               f"ra<{coordinates_transformed.ra.to_value(u.deg) + radius.to_value(u.deg) / np.cos(coordinates_transformed.dec)}")
+
+        tap_result = tap_service.run_sync(qstr)
+        tap_result = tap_result.to_table()
+        # filter out duplicated lines from the table
+        mask = tap_result['type'] != 'D'
+        filtered_table = tap_result[mask]
+
+        return filtered_table
+
+    def get_images(self, position, pixels=None, radius=0.5*u.arcmin, *, data_release=9, show_progress=True, image_band='g'):
+        """
+        Downloads the images for a certain region of interest.
+
+        Parameters
+        ----------
+        position: `~astropy.coordinates.SkyCoord`
+            coordinates around which we define our region of interest.
+        pixels: 'int', optional
+            pixel dimensions of the image to be produced.
+            Default to 60.
+        radius: `~astropy.units.Quantity`, optional
+            the radius of our region of interest.
+            Default to a value of 0.5 arcmin.
+        data_release: int, optional
+            the data release of the LegacySurvey to use.
+            Default to 9.
+        show_progress: bool, optional
+            whether to display a progress bar if the file is downloaded from a remote server.
+            Default to True.
+        image_band: str, optional
+            indicates the desired band.
+            Default to 'g'.
+
+        Returns
+        -------
+        list: A list of `~astropy.io.fits.HDUList` objects.
+        """
+
+        position_transformed = position.transform_to(coord.ICRS)
+
+        image_size_arcsec = radius.arcsec
+        pixsize = 2 * image_size_arcsec / pixels
+
+        image_url = (f"{conf.legacysurvey_service_url}?"
+                    f"ra={position_transformed.ra.deg}&"
+                    f"dec={position_transformed.dec.deg}&"
+                    f"size={pixels}&"
+                    f"layer=ls-dr{data_release}&"
+                    f"pixscale={pixsize}&"
+                    f"bands={image_band}")
+
+        file_container = commons.FileContainer(image_url, encoding='binary', show_progress=show_progress)
+
+        try:
+            fits_file = file_container.get_fits()
+        except (requests.exceptions.HTTPError, urllib.error.HTTPError) as exp:
+            fits_file = None
+            warnings.warn(f"{str(exp)} - Problem retrieving the file at the url: {image_url}", NoResultsWarning)
```

Comment:
Following @bsipocz 's unresolved comment: if the download was unsuccessful, we should raise an Exception.
```suggestion
            raise urllib.error.HTTPError(f"{str(exp)} - Problem retrieving the file at the url: {image_url}")
```